dind-agent/Dockerfile

```diff
@@ -1,5 +1,6 @@
-FROM docker:1.9-dind
-RUN apk --update add openjdk7-jre unzip git python python3 bash jq openssh-client
+FROM docker:1.10-dind
+# include dependencies needed by the jenkins agent itself
```

Comment:
Broke out the installs into two separate alphabetical lists. Ended up just having them as two separate RUNs for clarity's sake.